crates/tako/src/internal/common/resources/map.rs

```diff
@@ -8,22 +9,30 @@ pub const NVIDIA_GPU_RESOURCE_NAME: &str = "gpus/nvidia";
 pub const AMD_GPU_RESOURCE_NAME: &str = "gpus/amd";
 pub const MEM_RESOURCE_NAME: &str = "mem";
 
+pub(crate) type ResourceRqMap = Map<ResourceRqId, ResourceRequestVariants>;
+
 #[derive(Debug)]
-pub(crate) struct ResourceIdAllocator {
+pub(crate) struct GlobalResourceMapping {
+    resource_rq_from_id: ResourceRqMap,
+    resource_rq_to_id: Map<ResourceRequestVariants, ResourceRqId>,
```

Comment:
We could store the requests in a `Vec` and only keep indices in the hashmaps (something like `StableMap`). Although I doubt the performance change would be noticeable here.